lti_consumer/models.py

```diff
@@ -626,6 +663,24 @@ class LtiAgsLineItem(models.Model):
         blank=True
     )
 
+    # OIDC URL, Client ID and Deployment ID
+    # This allows us to isolate the LineItem per tool deployment
+    oidc_url = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+    client_id = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+    deployment_id = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+
```

Comment:
A single LtiConfiguration entity can only have one set of oidc_url, client_id, and deployment_id values set on the DB and XBlock, but once we use external configurations, the oidc_url, client_id and deployment_id values on the external configuration could change, this will mean that the gradebook won't change even if the oidc_url, client_id and deployment_id values changed for the LtiConfiguration. Since each lineitem should only expose gradebook information directly tied to the tool deployment, I modified the LtiAgsLineItem model to include the oidc_url, client_id and deployment_id so we are querying the LtiAgsLineItem on LtiAgsLineItemViewset not only by the lti_configuration but also by the values currently being used by the LtiConfiguration. I might be wrong about this, so please correct me if I'm wrong.

Also, you are correct, this fields should be filled in the migration, otherwise, this change wouldn't be backward compatible if applied.

We noticed the possibility of using deployment IDs while using an external configuration, I can separate anything related to implementing the use of multi-tenancy with external configurations on a separate PR to avoid increasing the complexity of the initial implementation of external configurations for LTI 1.3. What do you think if I add an ADR on how to set external configurations for LTI 1.3 and another ADR about external configurations multi-tenancy on this other PR I will open?.

Reply:
@kuipumu, yes, please move the multi-tenancy-related things to a separate PR. Once we merge the reusable LTI 1.3 configs, catching potential regressions will be much easier.

> What do you think if I add an ADR on how to set external configurations for LTI 1.3 and another ADR about external configurations multi-tenancy on this other PR I will open?.

Sounds reasonable to me. Do we have a rough timeline for this separate PR? It would be good to add the LTI 1.3 external configurations ADR before the Quince release.

Reply:
@kuipumu, Quince will be released [in December](https://docs.google.com/spreadsheets/d/11DheEtMDGrbA9hsUvZ2SEd4Cc8CaC4mAfoV8SVaLBGI/edit#gid=195838733&range=AV5:BA5).